avocado/core/task/runtime.py

```diff
@@ -63,14 +64,34 @@ def __init__(self, test_suite, job_id, status_server_uri):
         self.test_suite = test_suite
         self.job_id = job_id
         self.status_server_uri = status_server_uri
+        self.graph = {}
 
-    def _get_requirements_runtime_tasks(self, runnable, prefix):
+    def _add_edge(self, pre_runtime_task, post_runtime_task):
+        post_runtime_task.dependencies.append(pre_runtime_task)
+
+    def _topological_order(self):
+        def topological_order_util(vertex, visited, topological_order):
+            visited[vertex] = True
+            for v in vertex.dependencies:
+                if not visited[v]:
+                    topological_order_util(v, visited, topological_order)
+            topological_order.append(vertex)
+
+        visited = dict.fromkeys(self.graph, False)
+        topological_order = []
+
+        for vertex in self.graph.values():
+            if not visited[vertex]:
+                topological_order_util(vertex, visited, topological_order)
+        return topological_order
+
+    def _create_requirements_runtime_tasks(self, runtime_task, prefix):
```

Comment:
Before, this method it was *only* creating the runtime tasks, nothing inplace. Now, you are also updating the graph and edges inplace. It would be nice to make it clear at the method name (probably choosing a new one) or spiting this method into two. Improving the code readability.